src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs

```diff
@@ -66,7 +66,10 @@ protected override void AnalyzeCore(Uri uri, string text, string solutionDirecto
             using (context)
             {
                 cancellationToken.ThrowIfCancellationRequested();
-                AnalyzeCommand.AnalyzeTargetHelper(context, this.rules, disabledSkimmers);
+
+                // Filtering file before analyzing.
+                IEnumerable<Skimmer<AnalyzeContext>> applicableSkimmers = AnalyzeCommand.DetermineApplicabilityForTargetHelper(context, this.rules, disabledSkimmers);
+                AnalyzeCommand.AnalyzeTargetHelper(context, applicableSkimmers, disabledSkimmers);
```

Comment:
@michaelcfanning , I was trying to simulate a file where it does not finish with the expected extension, with that, we shouldn't show error.
The first thing that I noticed is that based on the unittests, we have to filter (this change does that).
The second thing: what is the difference between `DefaultNameRegex` and `NameRegex`? I saw that the filter is just applied for the `DefaultNameRegex`.
Thank you :)